File: 13_CRUD_MultipleTables_EJ3-1/src/main/java/com/example/_Database_DB1/Persona/domain/persona.java
package com.example._Database_DB1.Persona.domain;


import com.example._Database_DB1.Student.domain.profesor;
import com.example._Database_DB1.Student.domain.student;
import lombok.Data;


import javax.persistence.*;
import java.util.Date;

@Entity
@Data
public class persona {
    @Id
    @GeneratedValue
    @Column(name = "ID_PERSONA")
    private int id_persona;

    private String usuario;
    private String password;
    private String name;
    private String surname;
    private String company_email;
    private String personal_email;
    private String city;
    private Boolean active;

    @Temporal(TemporalType.DATE)
    private Date created_date;

    private String imagen_url;

    @Temporal(TemporalType.DATE)
    private Date termination_date;

    @OneToOne
    @JoinColumn(name = "ID_STUDENT")
    @MapsId
    private com.example._Database_DB1.Student.domain.student student;
/*
    @OneToOne
    @JoinColumn(name = "ID_PROFESOR")
    @MapsId
    private com.example._Database_DB1.Student.domain.profesor profesor;

 */
}

File: frontend/angular_Crud_menu_guards_BD_resolver/src/app/personas/persona.ts

export interface Persona {
    position: number;

    name: string;
    //password: String ;
    usuario: string;
    //surname: String ;
    emailC: string;
    emailP: string;
    city: string;

    act: boolean;

    DateInit: string;
    url: string;
    DateEnd: string;
}

export interface PersonaHttp {
    position: number;

    name: string;
    password: String ;
    usuario: string;
    surname: String ;
    company_email: string;
    personal_email: string;
    city: string;

    active: boolean;

    created_date: String;
    imagen_url: string;
    termination_date: String;
}




export class User {
    constructor(
        public name: string,
        public position: number,
        public usuario: string,
        public emailC: string,
        public emailP: string,
        public city: string,
        public url: string,
        public DateInit: String,
        public act: boolean,
        public DateEnd: String
    ){}
}
  

export class UserFetch {
    constructor( public id = 0, public name = '', public model='') {}
}
  
  